File: Springboot_API/RetaCantabria/src/main/java/org/example/Entidades/Resena.java
package org.example.Entidades;

import jakarta.persistence.*;

import java.time.LocalDate;

@Entity
@Table(name="resenas")
public class Resena {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int idResena;
    @Column(name = "resena",columnDefinition = "TEXT")
    private String resena;
    private LocalDate fecha;
    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "idUsuario", nullable = false)
    private Usuario usuario;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "idRuta", nullable = false)
    private Ruta ruta;

}

File: AplicacionMovil/app/src/main/java/com/example/retodam2rutas/data/entities/Resena.kt
package com.example.retodam2rutas.data.entities

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.ForeignKey
import androidx.room.Index
import androidx.room.PrimaryKey
import java.time.LocalDate

@Entity(
    tableName = "resenas",
    foreignKeys = [
        ForeignKey(
            entity = Usuario::class,
            parentColumns = ["idUsuario"],
            childColumns = ["usuarioId"],
            onDelete = ForeignKey.CASCADE
        ),
        ForeignKey(
            entity = Ruta::class,
            parentColumns = ["idRuta"],
            childColumns = ["rutaId"],
            onDelete = ForeignKey.CASCADE
        )
    ],
    indices = [
        Index(value = ["usuarioId"]),
        Index(value = ["rutaId"])
    ]
)
data class Resena(

    @PrimaryKey(autoGenerate = true)
    @ColumnInfo(name = "idResena")
    val idResena: Int = 0,

    @ColumnInfo(name = "resena")
    val resena: String,

    @ColumnInfo(name = "fecha")
    val fecha: LocalDate,

    @ColumnInfo(name = "usuarioId")
    val usuarioId: Int,

    @ColumnInfo(name = "rutaId")
    val rutaId: Int
)
